### PR TITLE
Fix firefox bug by detecting invalid xml and document response type

### LINKED ISSUE #111: 204 (no content) responses return an XMLDocument in Firefox
When the server returns a 204, firefox sets `responseXML` to an XML document - a truthy object.

This is apparently a [long-standing Firefox bug](https://bugzilla.mozilla.org/show_bug.cgi?id=521301).

Since `xhr` is abstracting away `response`/`responseText`/`responseXML`, this means that `xhr` returns something other than null/undefined/empty string in Firefox when the server returns a 204.

https://github.com/Raynos/xhr/blob/master/index.js#L74

Would you be open to a pull request to fix this?  How would you prefer that this was fixed?  A check for a 204 status code that always returns null/undefined?  Or maybe return `responseText` as long as it's a string, so that an empty string is returned?

I've opened a pull request with a failing test: #112

### PR DESCRIPTION
**Pull request by TehShrike**
Based on #118 and inspired by @ciscoheat in #120.

Fixes #111

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -68,8 +68,9 @@ function _createXHR(options) {
 
         if (xhr.response) {
             body = xhr.response
-        } else if (xhr.responseType === "text" || !xhr.responseType) {
-            body = xhr.responseText || xhr.responseXML
+        } else {
+            var xml = getXml(xhr)
+            body = xhr.responseText || xml
```
Comment (naugtur) [inline index.js]: if xhr.responseText is available, all work done by getXml is for nothing. 
If you roll it back to a single line, the function will only be called if needed.

Comment (TehShrike) [inline index.js]: Good call.

Comment (TehShrike) [inline index.js]: I pushed that improvement up.

**Comment by naugtur** [comment]
Good solution overall, thanks for holding on despite my limited activity recently.

Let me run the tests and I'll merge it while I'm here ;)

**Comment by naugtur** [comment]
I'll do some housekeeping and release a new version this week.

**Comment by TehShrike** [comment]
Right on!

File: index.js [hunk 2]
```diff
@@ -216,4 +216,16 @@ function _createXHR(options) {
 
 }
 
+function getXml(xhr) {
+    if (xhr.responseType === "document") {
+        return xhr.responseXML
+    }
+    var firefoxBugTakenEffect = xhr.status === 204 && xhr.responseXML && xhr.responseXML.documentElement.nodeName === "parsererror"
```
Comment (Mouvedia) [inline index.js]: This is too restrictive. 201, 202 and 205 are also affected.
cf https://bugzilla.mozilla.org/show_bug.cgi?id=884693#c34

**Pull request merged**